### ISSUE TITLE
ListType field selection fails when marked as NonNullType<ListType<>>

### ISSUE THREAD

**Issue by JakeSmokie** [ChilliCream/hotchocolate#2568]
So oddly this code fails

```cs
descriptor.Field("some_ints")
                .Type<NonNullType<ListType<NonNullType<IntType>>>>()
                .Resolver(new List<int> { 1, 2, 3 });
```

```
A resolver error occured
System.InvalidOperationException: Operation is not valid due to the current state of the object.
   at HotChocolate.Data.Projections.Expressions.Handlers.ExpressionExtensions.Append(Expression expression, MemberInfo memberInfo)
   at HotChocolate.Data.Projections.Expressions.Handlers.QueryableProjectionListHandler.OnBeforeEnter(QueryableProjectionContext context, ISelection selection)
   at HotChocolate.Data.Projections.ProjectionVisitor`1.OnBeforeEnter(ISelection selection, TContext context)
   at HotChocolate.Data.Projections.SelectionVisitor`1.Visit(ISelection selection, TContext context)
   at HotChocolate.Data.Projections.ProjectionVisitor`1.Visit(ISelection selection, TContext context)
   at HotChocolate.Data.Projections.SelectionVisitor`1.VisitChildren(IOutputField field, TContext context)
   at HotChocolate.Data.Projections.SelectionVisitor`1.Visit(IOutputField field, TContext context)
   at HotChocolate.Data.Projections.ProjectionVisitor`1.Visit(IOutputField field, TContext context)
   at HotChocolate.Data.Projections.ProjectionVisitor`1.Visit(TContext context)
   at HotChocolate.Data.Projections.Expressions.QueryableProjectionProvider.<CreateExecutor>g__ExecuteAsync|2_1[TEntityType](FieldDelegate next, IMiddlewareContext context)
   at HotChocolate.Execution.Processing.ResolverTask.ExecuteResolverPipelineAsync(CancellationToken cancellationToken)
   at HotChocolate.Execution.Processing.ResolverTask.TryExecuteAsync(CancellationToken cancellationToken)
```

Without type definition everything works fine, but field has type `[Int!]`
I use `HotChocolate 11.0.0-rc1`, and C# reference nullables are enabled

**Comment by JakeSmokie** [comment]
A little remark: here is my query code
```cs
        [UseProjection]
        [UseFiltering]
        public IQueryable<Item> GetItems([Service] ItemContext context) => 
            context.Items.AsNoTracking();
```

**Comment by JakeSmokie** [comment]
Just found out that projection code is executed if you call `Type<>()` for `IObjectTypeDescriptor`.

Calling `.IsProjected(false)` helps in this case
```
descriptor.Field("some_ints")
                .Type<NonNullType<ListType<NonNullType<IntType>>>>()
                .Resolver(new List<int> { 1, 2, 3 })
                .IsProjected(false);
```

**Comment by michaelstaib** [comment]
@PascalSenn can you check if this needs to be fixed for 11.0?
